recipes-kernel/linux/linux-renesas/0001-Add-device-driver-for-rcar-r7-rpmsg.patch

```diff
@@ -0,0 +1,143 @@
++	  .
++	  .
++	  .
++
+ config RPMSG_QCOM_GLINK_NATIVE
+ 	tristate
+ 	select RPMSG
+diff --git a/drivers/rpmsg/rpmsg_char.c b/drivers/rpmsg/rpmsg_char.c
+index eea5ebbb5119..aabc3fc4b3c2 100644
+--- a/drivers/rpmsg/rpmsg_char.c
++++ b/drivers/rpmsg/rpmsg_char.c
+@@ -123,12 +123,15 @@ static int rpmsg_ept_cb(struct rpmsg_device *rpdev, void *buf, int len,
+ static int rpmsg_eptdev_open(struct inode *inode, struct file *filp)
+ {
+ 	struct rpmsg_eptdev *eptdev = cdev_to_eptdev(inode->i_cdev);
++#ifndef CONFIG_RPMSG_RCAR_CHAR
+ 	struct rpmsg_endpoint *ept;
+ 	struct rpmsg_device *rpdev = eptdev->rpdev;
++#endif /* CONFIG_RPMSG_RCAR_CHAR */
+ 	struct device *dev = &eptdev->dev;
+ 
+ 	get_device(dev);
+ 
++#ifndef CONFIG_RPMSG_RCAR_CHAR
+ 	ept = rpmsg_create_ept(rpdev, rpmsg_ept_cb, eptdev, eptdev->chinfo);
+ 	if (!ept) {
+ 		dev_err(dev, "failed to open %s\n", eptdev->chinfo.name);
+@@ -137,6 +140,7 @@ static int rpmsg_eptdev_open(struct inode *inode, struct file *filp)
+ 	}
+ 
+ 	eptdev->ept = ept;
++#endif /* CONFIG_RPMSG_RCAR_CHAR */
+ 	filp->private_data = eptdev;
+ 
+ 	return 0;
+@@ -338,21 +342,35 @@ static void rpmsg_eptdev_release_device(struct device *dev)
+ 	kfree(eptdev);
+ }
+ 
++#ifndef CONFIG_RPMSG_RCAR_CHAR
+ static int rpmsg_eptdev_create(struct rpmsg_ctrldev *ctrldev,
+ 			       struct rpmsg_channel_info chinfo)
+ {
+ 	struct rpmsg_device *rpdev = ctrldev->rpdev;
++#else
++static int rpmsg_eptdev_create(struct rpmsg_device *rpdev)
++{
++#endif /* CONFIG_RPMSG_RCAR_CHAR */
+ 	struct rpmsg_eptdev *eptdev;
+ 	struct device *dev;
+ 	int ret;
+-
+ 	eptdev = kzalloc(sizeof(*eptdev), GFP_KERNEL);
+ 	if (!eptdev)
+ 		return -ENOMEM;
+ 
+ 	dev = &eptdev->dev;
+ 	eptdev->rpdev = rpdev;
++#ifndef CONFIG_RPMSG_RCAR_CHAR
+ 	eptdev->chinfo = chinfo;
++#else
++	memcpy(eptdev->chinfo.name, rpdev->id.name, RPMSG_NAME_SIZE);
++	eptdev->chinfo.name[RPMSG_NAME_SIZE-1] = '\0';
++
++	eptdev->chinfo.src = rpdev->src;
++	eptdev->chinfo.dst = rpdev->dst;
++	eptdev->ept = rpdev->ept;
++	eptdev->ept->priv=eptdev;
++#endif /* CONFIG_RPMSG_RCAR_CHAR */
+ 
+ 	mutex_init(&eptdev->ept_lock);
+ 	spin_lock_init(&eptdev->queue_lock);
+@@ -361,7 +379,11 @@ static int rpmsg_eptdev_create(struct rpmsg_ctrldev *ctrldev,
+ 
+ 	device_initialize(dev);
+ 	dev->class = rpmsg_class;
++#ifndef CONFIG_RPMSG_RCAR_CHAR
+ 	dev->parent = &ctrldev->dev;
++#else
++	dev->parent = &rpdev->dev;
++#endif /* CONFIG_RPMSG_RCAR_CHAR */
+ 	dev->groups = rpmsg_eptdev_groups;
+ 	dev_set_drvdata(dev, eptdev);
+ 
+@@ -424,6 +446,7 @@ static int rpmsg_ctrldev_release(struct inode *inode, struct file *filp)
+ 	return 0;
+ }
+ 
++#ifndef CONFIG_RPMSG_RCAR_CHAR
+ static long rpmsg_ctrldev_ioctl(struct file *fp, unsigned int cmd,
+ 				unsigned long arg)
+ {
+@@ -544,6 +567,22 @@ static struct rpmsg_driver rpmsg_chrdev_driver = {
+ 		.name = "rpmsg_chrdev",
+ 	},
+ };
++#else
++static struct rpmsg_device_id rpmsg_driver_sample_id_table[] = {
++	{ .name	= "rpmsg-client-sample" },
```

Comment:
Adding rpmsg-client-sample as compatible of this driver make it conflicting with the SAMPLE_RPMSG_CLIENT, one way of solving that could be to change the rpmsg endpoint and use the new endpoint name that could be "rpmsg-rcar-char-sample".
So that depending on the zephyr firmware loaded into the MCU both driver are possible either the in kernel space one or the userland one.